### PR TITLE
port compute_diag_third_shoc_moment

### PR DESCRIPTION
**Pull request by tcclevenger**
Port `compute_diag_third_shoc_moment` to C++.

This pull request was already reviewed here: https://github.com/E3SM-Project/scream/pull/602. I decided to switch to non-forked branches so the tester could run without approval, otherwise same code.

### CODE REVIEW

**Review by ambrad — APPROVED** [review]

**Comment by tcclevenger** [comment]
@ambrad because I had to fix conflicts and merge master, I need another approval before it can merge.

**Review by ambrad — APPROVED** [review]

**Pull request merged**
